email/provider_test.go

```diff
@@ -79,34 +85,41 @@ func TestRetryWorkflow(t *testing.T) {
 	maxRetries := 5
 	// write the tests
 	tests := []struct {
-		name         string
-		sendMailFunc SendMailFunc
-		expectedErr  error
+		name          string
+		sendMailFunc  SendMailFunc
+		expectedErr   error
+		expectedCount int
 	}{
 		{
-			name:         "Retry with 421 error",
-			sendMailFunc: mockSendMailReturn421,
-			expectedErr:  errors.New("421 Service not available"),
+			name:          "Retry with 421 error",
+			sendMailFunc:  mockSendMailReturn421,
+			expectedErr:   errors.New("421 Service not available"),
+			expectedCount: maxRetries,
 		},
 		{
-			name:         "Success on first try",
-			sendMailFunc: mockSendMailSuccess,
-			expectedErr:  nil,
+			name:          "Success on first try",
+			sendMailFunc:  mockSendMailSuccess,
+			expectedErr:   nil,
+			expectedCount: 1,
 		},
 		{
-			name:         "Other error",
-			sendMailFunc: mockReturn500,
-			expectedErr:  errors.New("500 Internal Server Error"),
+			name:          "Other error",
+			sendMailFunc:  mockReturn500,
+			expectedErr:   errors.New("500 Internal Server Error"),
+			expectedCount: 1,
 		},
 	}
 	// execute the tests
 	for _, test := range tests {
 		// run subtests
 		t.Run(test.name, func(t *testing.T) {
+			// set test retries zero before each test
+			testRetries = 0
 			err := sendMail(test.sendMailFunc, maxRetries, "localhost", "2525", "username", "password", "from@example.com", "to@example.com", "message")
 			if test.expectedErr != nil {
 				// assert that the errors are equal
 				assert.EqualError(t, err, test.expectedErr.Error())
+				assert.Equal(t, testRetries, test.expectedCount)
```

Comment:
We can check this when there are no errors too

Reply:
whoops, not sure how I missed that. Thank you